### PR TITLE
[react-native-unimodules] Remove `expo-permissions` from dependencies.

### PR DESCRIPTION
**Pull request by bbarthec**
`expo-permissions` is now deprecated — the functionality has been moved to other expo packages that directly use these permissions.

# Why

It's no longer bundled in the `expo` package, and it's also deprecated.

# How

Removed `expo-permissions` from dependencies in `package.json`.
Removed re-exported `Permissions` object.

# Test Plan

- [x] run `ncl` on UNVERSIONED with no problem on iOS and Android

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/react-native-unimodules/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 ### 🛠 Breaking changes
 
+- Removed exported `Permissions` object and removed `expo-permissions` from dependencies. ([#TO_BE_FILLED](https://github.com/unimodules/react-native-unimodules/pull/TO_BE_FILLED) by [@bbarthec](https://github.com/bbarthec))
```
Comment (tsapeta) [inline packages/react-native-unimodules/CHANGELOG.md:7]:
```suggestion
- Removed exported `Permissions` object and removed `expo-permissions` from dependencies. ([#12405](https://github.com/unimodules/react-native-unimodules/pull/12405) by [@bbarthec](https://github.com/bbarthec))
```

**Review by tsapeta — APPROVED** [review]

**Comment by leethree** [comment]
looks like it's added back in bb3887cf44b45e3fdb262fdae31ad7fc79d504b7 . is this intentional?

https://github.com/expo/expo/blob/bb3887cf44b45e3fdb262fdae31ad7fc79d504b7/packages/react-native-unimodules/package.json#L46

**Comment by tsapeta** [comment]
cc @brentvatne ☝️

**Comment by brentvatne** [comment]
ah, it looks like this was never cherry-picked to the sdk-41 branch and so when we published from that branch and cherry-picked, this happened...

**Pull request merged**